docs/src/main/sphinx/glossary.rst

```diff
@@ -0,0 +1,187 @@
+========
+Glossary
+========
+
+The glossary contains a list of key Trino terms and definitions.
+
+.. _glossCatalog:
+
+Catalog
+    Catalogs define and name a configuration for connecting to a data source,
+    allowing users to query the connected data. Each catalog's configuration
+    specifies a :ref:`connector <glossConnector>` to define which data source
+    the catalog connects to. For more information, see
+    :ref:`trino-concept-catalog`.
```

Comment:
Use this format or similar, like "For more information about X, see `link`" elsewhere in this doc to be consistent. The Container and Data Source entries have similar links that should be introduced the same way as this.